### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by Xennis**
See commit messages

### CODE REVIEW

File: doc/API-mapping.md [hunk 1]
```diff
@@ -171,6 +171,33 @@ Now, the API call URLs will look like:
 
 http://your.api.domain/entityName?q=foo&tag=bar
 
+
+## Nested request and response
+
+If your API returns the results in a field of an object instead directly, you can use Restangular response interceptor. For instance if you request http://your.api.domain/entityName and the API returns the result array in an `items` field instead directly:
+
+```js
+app.config(function(RestangularProvider) {
```
Comment (fzaninotto) [inline doc/API-mapping.md]: This is already covered in the 'Entry Format', line 40.

File: doc/API-mapping.md [hunk 2]
```diff
@@ -171,6 +171,33 @@ Now, the API call URLs will look like:
...
+    RestangularProvider.addResponseInterceptor(function(data, operation, what, url, response) {
+        if (operation == "getList" && what == 'entityName') {
+            data = data.items;
+        }
+        return data;
+});
+```
+
+If your API requires that you post and put data for instance inside of an `item` field instead directly, use Restangular request interceptor:
```
Comment (fzaninotto) [inline doc/API-mapping.md]: Good, but you shouldn't add a new section about it, "Entry Format" is there for it.

**Comment by Xennis** [comment]
> This is already covered in the 'Entry Format', line 40.

In line 40 the method with the `RestangularProvider.addElementTransformer` is explained, but not the method with the `RestangularProvider.addResponseInterceptor`, when your data looks like that:

```
GET http://your.api.domain/entityName
{
  meta: { totalCount: 40, page: 2, ... },
  items: { HERE IS THE DATA ARRAY }
}
```

. The response interceptor is used in the "Total Number of Results" section too, so you can combine both easily.


> shouldn't add a new section about it, "Entry Format" is there for it.

Okay

**Comment by fzaninotto** [comment]
When your data looks like the code extract you pasted, this is exactly the case to use  `addElementTransformer` instead of `addResponseInterceptor`. Besides, I don't want to replicate the entire Restangular documentation in ng-admin.

**Comment by Xennis** [comment]
Okay, good to know.

**Comment by Xennis** [comment]
Okay, good to know.

So may you just merge my first commit.

**Comment by fzaninotto** [comment]
I'm interested in the second part of the Mapping addition, about ResponseInterceptor. Can you please update your PR accordingly?

**Comment by Xennis** [comment]
Of course. But I am confused, you wrote
> this is exactly the case to use addElementTransformer instead of add**ResponseInterceptor**

and now
>  second part of the Mapping addition, about **ResponseInterceptor**

Was is just a typo and you meant, that you want the part about the RequestInterceptor?

**Comment by fzaninotto** [comment]
Yes, it was a typo, sorry. We don't need the addition about response interceptor, but we need the addition about FullRequestInterceptor.

**Comment by Xennis** [comment]
Okay, here we go.

**Comment by fzaninotto** [comment]
Sorry, needs rebase

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**